docs/source/expression.rst

```diff
@@ -0,0 +1,158 @@
+.. Copyright (c) 2016, Johan Mabille and Sylvain Corlay
+
+   Distributed under the terms of the BSD 3-Clause License.
+
+   The full license is in the file LICENSE, distributed with this software.
+
+Expressions and lazy evaluation
+===============================
+
+`xtensor` is more than an N-dimensional array library: it's an expression engine that allows numerical computation on any object implementing the expression interface.
+These objects can be in-memory containers such as ``xarray<T>`` and ``xtensor<T>``, but can also be backed by a database or a representation on the file system. This
+also enables creating adaptors as expressions for other data structures.
+
+Expressions
+-----------
+
+Assume ``x``, ``y`` and ``z`` are arrays of *broadcastable shapes* (we'll come back to that later), the return type of an expression such as ``x + y * sin(z)`` is **not an array**.
+It is an ``xexpression`` object offering the same interface as an N-dimensional array. You can store the result of such an expression and reuse it to build even more complex
```

Comment:
The result is an `xexpression` which offers the same interface as an N-dimensional array but does not hold any value. Such expressions can be plugged into others to build more complex expressions.